File: piwigo/Login/UIComponents/PiwigoTextField.m
//
//  PiwigoTextField.m
//  piwigo
//
//  Created by Spencer Baker on 1/17/15.
//  Copyright (c) 2015 bakercrew. All rights reserved.
//

#import "PiwigoTextField.h"

@implementation PiwigoTextField

-(instancetype)init
{
	self = [super init];
	if(self)
	{
		self.layer.cornerRadius = 10.0;
		self.font = [UIFont piwigoFontNormal];
        self.translatesAutoresizingMaskIntoConstraints = NO;
        self.clearButtonMode = YES;
        self.autocapitalizationType = UITextAutocapitalizationTypeNone;
        self.autocorrectionType = UITextAutocorrectionTypeNo;
        self.keyboardAppearance = [Model sharedInstance].isDarkPaletteActive ? UIKeyboardAppearanceDark : UIKeyboardAppearanceDefault;
	}
	return self;
}

-(CGRect)textRectForBounds:(CGRect)bounds
{
	return CGRectInset(bounds, 10, 0);
}

-(CGRect)editingRectForBounds:(CGRect)bounds
{
	return CGRectInset(bounds, 10, 0);
}

@end

File: piwigo/Login/UIComponents/PiwigoTextField.swift
//
//  PiwigoTextField.swift
//  piwigo
//
//  Created by Spencer Baker on 1/17/15.
//  Copyright (c) 2015 bakercrew. All rights reserved.
//
//  Converted to Swift 5.4 by Eddy Lelièvre-Berna on 26/02/2022.
//

import UIKit

class PiwigoTextField: UITextField {
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
        layer.cornerRadius = 8.0
        font = UIFont.piwigoFontNormal()
        translatesAutoresizingMaskIntoConstraints = false
        clearButtonMode = UITextField.ViewMode.always
        autocapitalizationType = .none
        autocorrectionType = .no
        keyboardAppearance = AppVars.shared.isDarkPaletteActive ? .dark : .default
    }

    override func textRect(forBounds bounds: CGRect) -> CGRect {
        return bounds.insetBy(dx: 10, dy: 0)
    }

    override func editingRect(forBounds bounds: CGRect) -> CGRect {
        return bounds.insetBy(dx: 10, dy: 0)
    }

    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
}
